src/index.ts

```diff
@@ -130,7 +144,22 @@ function makeSourceMapAndFinish(
   );
 
   setModuleMeta(loaderContext, instance, fileVersion);
-  callback(null, output, sourceMap);
+
+  // if we don't have an input source map, we only need to return the newly generated source map
+  // this avoids that we have to make a possibly expensive call to the source-map lib
+  if (inputSourceMap === undefined) {
```

Comment:
Also `sourceMap` could potentially be undefined if the ts compiler is not configured to generate source maps. 

This is a bit confusing because the developer has to configure source maps in two places: The webpack config and also the tsconfig. I'm wondering if the property `webpack.LoaderContext.sourceMap` indicates whether webpack would like to have source maps generated or not. If yes, then I guess it makes sense to issue a warning to the user when the ts config source map option and the webpack source map config differ. @johnnyreilly  What do you think?

Reply:
Potentially yes, let's come back to this later though

Reply:
Okay, I've fixed the bug where `sourceMap` being undefined would crash ts-loader and then we can later concentrate on the warning.